prime-router/src/main/kotlin/azure/BlobAccess.kt

```diff
@@ -356,6 +360,26 @@ class BlobAccess() : Logging {
             return toBlobUrl
         }
 
+        fun restorePreviousVersion(
+            blobItemAndPreviousVersions: BlobItemAndPreviousVersions,
+            blobConnInfo: BlobContainerMetadata,
+        ) {
+            val container = getBlobContainer(blobConnInfo)
+            val currentBlobClient = container.getBlobClient(blobItemAndPreviousVersions.currentBlobItem.name)
+            val previousVersionId = blobItemAndPreviousVersions.previousBlobItemVersions?.firstOrNull()?.versionId
+            if (previousVersionId != null) {
+                val previousVersionBlobClient = container.getBlobVersionClient(
+                    blobItemAndPreviousVersions.currentBlobItem.name,
+                    previousVersionId
+                )
+                currentBlobClient.copyFromUrl(previousVersionBlobClient.blobUrl)
+            } else {
+                logger.error(
```

Comment:
I toyed with throwing and exception here but ultimately decided it was not necessary and there wasn't anything to recover.